lib/iex/lib/iex/autocomplete.ex

```diff
@@ -643,4 +647,90 @@ defmodule IEx.Autocomplete do
   defp extract_from_ast(_ast_node, _acc) do
     :error
   end
+
+  defp path_fragment(expr), do: collect_path_fragment(expr, [])
+
+  # collects the fragment and store it on `acc`, calls `return_path_fragment` 
+  # when it finds a `"`, discards `acc` otherwise
+  defp collect_path_fragment([], _acc), do: []
+  defp collect_path_fragment([?", ?\\ | t], acc), do: collect_path_fragment(t, [?\\, ?" | acc])
+  defp collect_path_fragment([?{, ?# | _rest], _acc), do: []
+  defp collect_path_fragment([?" | t], acc), do: return_path_fragment(t, acc)
```

Comment:
Another way to actually check if the collected fragment is an open string would be to check `rem(Enum.count(t, & &1 == ?"), 2) == 0`. Do you think that would be better?

Reply:
Actually, nevermind, just realized that would not work because we have to skip the escaped `"`